qupulse/program/transformation.py

```diff
@@ -88,7 +88,16 @@ def get_constant_output_channels(self, input_channels: AbstractSet[ChannelID]) -
 
 class ChainedTransformation(Transformation):
     def __init__(self, *transformations: Transformation):
-        self._transformations = transformations
+        #avoid nesting also here in init to ensure always flat hierachy?
```

Comment:
This is the job of the function `chain_transformations`

Reply:
yes; basically copied over that code since the function instantiates a new ChainedTransformation.
I was'nt sure if ChainedTransformation is called directly anywhere instead of being instantiated with .chain, so I did the check again in the init function. If the class is not supposed to be initialized directly, it could also be removed again.